frontend/src/apis/joinCompetition/index.ts

```diff
@@ -0,0 +1,48 @@
+import api from '@/utils/api';
+
+import type { CompetitionApiData } from './types';
+import axios from 'axios';
+
+const STATUS = {
+  Forbidden: 403,
+} as const;
+
+export async function joinCompetition(data: CompetitionApiData) {
+  const { id, token } = data;
+
+  try {
+    await api.post(
+      `/competitions/${id}/participations`,
+      {},
+      {
+        headers: {
+          Authorization: `Bearer ${token}`,
+        },
+      },
+    );
+
+    return '대회에 성공적으로 참여했습니다.';
+  } catch (error: unknown) {
+    let errorMessage = 'Unexpected error occurred.';
+
+    if (axios.isAxiosError(error)) {
+      if (error.response) {
+        switch (error.response.status) {
+          case STATUS.Forbidden:
+            errorMessage = '대회 참여에 실패했습니다. 서버에서 거절되었습니다.';
+            break;
+          case 400:
+            errorMessage = '이미 참여한 대회입니다.';
+            break;
+          default:
+            errorMessage = `HTTP Error ${error.response.status}`;
+            break;
+        }
+      } else {
+        errorMessage = 'Network error occurred.';
+      }
+    }
```

Comment:
isAxiosError 사용한 부분 아주 좋습니다. :) 제 개인적인 의견으로는 가독성을 위해

```suggestion
    if (!axios.isAxiosError(error)) {
      return 'Unexpected error occurred';
    } 
    
    if (!error.response) {
      return 'Network error occurred';
    }
    
    switch(error.response.status) {
      case STATUS.Forbidden:
            errorMessage = '대회 참여에 실패했습니다. 서버에서 거절되었습니다.';
            break;
      case 400:
            errorMessage = '이미 참여한 대회입니다.';
            break;
      default:
            errorMessage = `HTTP Error ${error.response.status}`;
            break;
      }
```

같은 코드가 어때요?

그리고 에러를 다양하게 핸들링한 부분 좋았어요. 기왕이면 STATUS.BadRequest도 사용하면 좋을 것 같네요

Reply:
감사합니다!